src/Common/Core/Impl/Services/ServiceManager.cs

```diff
@@ -40,24 +33,36 @@ public class ServiceManager : IServiceManager {
             type = type ?? typeof(T);
             Check.ArgumentNull(nameof(service), service);
             Check.InvalidOperation(() => _s.TryAdd(type, service), "Service already exists");
-            ServiceAdded?.Invoke(this, new ServiceContainerEventArgs(type));
+
+            var mainThread = service as IMainThread;
```

Comment:
There is code that wants main thread *after* SM is disposed... Mostly tests though (fixture gets disposed BEFORE test actually completes) but I guess there could be cases (idle, async) when call completes after disposal.